File: testsuite/tests/C/mcdc/sign-warn/src/foo.c
#include "foo.h"

#include <stddef.h>

int
do_test (int *x, int *y, int *z)
{

  // pointer equality ops in condition
  if (x == NULL || y == NULL) // # ob
    {

      // and in decision
      if (z == NULL) // # ob
        return 0;    // # ob
      return *z;     // # ob
    }

  //  Relational ops in conditions
  if (*x < 0 && *y < 0) // # ob
    return *z;          // # ob

  // Relational ops in decision
  if (*y < 0)  // # ob
    return *y; // # ob

  return *z; // # ob
}

File: testsuite/tests/llvm/rust/stmt/if-expr/src/foo.rs
use std::hint::black_box;

pub fn foo(b: bool) { // # decl-1
    if b              // # condition
    {                 // # if-1
        black_box(1); // # if
    }                 // # if
    else
    {                 // # else-1
        black_box(2); // # else
    }                 // # else
}                     // # end-1
